--- a/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php
+++ b/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php
@@ -30,8 +30,8 @@ public function isCompatible(FieldItemListInterface $items) {
     // This editor is compatible with formatted ("rich") text fields; but only
     // if there is a currently active text format, that text format has an
     // associated editor and that editor supports inline editing.
-    elseif ($editor = editor_load($items[0]->format)) {
+    elseif (is_object($items[0]) && $editor = editor_load($items[0]->format)) {
     elseif (is_object($items[0]) && $editor = editor_load($items[0]->format)) {
       $definition = \Drupal::service('plugin.manager.editor')->getDefinition($editor->getEditor());
       if ($definition['supports_inline_editing'] === TRUE) {
         return TRUE;
