diff --git a/modules/contrib/group/src/Entity/Controller/GroupContentController.php b/modules/contrib/group/src/Entity/Controller/GroupContentController.php
index 4b4c075d5..f328af71d 100644
--- a/modules/contrib/group/src/Entity/Controller/GroupContentController.php
+++ b/modules/contrib/group/src/Entity/Controller/GroupContentController.php
@@ -376,7 +376,8 @@ public function createFormTitle(GroupInterface $group, $plugin_id) {
     /** @var \Drupal\group\Plugin\GroupContentEnablerInterface $plugin */
     $plugin = $group->getGroupType()->getContentPlugin($plugin_id);
     $group_content_type = GroupContentType::load($plugin->getContentTypeConfigId());
-    return $this->t('Add @name', ['@name' => $group_content_type->label()]);
+    return $this->t('@name', ['@name' => $group_content_type->label()]);
+    //return $this->t('Add @name', ['@name' => $group_content_type->label()]);
   }
 
 }
