diff --git a/modules/contrib/group/src/Entity/GroupContent.php b/modules/contrib/group/src/Entity/GroupContent.php
index b3ca633..e67119e 100644
--- a/modules/contrib/group/src/Entity/GroupContent.php
+++ b/modules/contrib/group/src/Entity/GroupContent.php
@@ -205,7 +205,7 @@ public function postSave(EntityStorageInterface $storage, $update = TRUE) {
       // flush some caches containing the entity or perform other operations we
       // cannot possibly know about. Lucky for us, all of that behavior usually
       // happens when saving an entity so let's re-save the added entity.
-      $this->getEntity()->save();
+//      $this->getEntity()->save();
     }
   }
 
