diff --git a/src/Plugin/EntityPrint/PrintEngine/DomPdf.php b/modules/entity_print/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
index 5631d403..9396c129 100644
--- a/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
+++ b/src/Plugin/EntityPrint/PrintEngine/DomPdf.php
@@ -153,7 +153,7 @@ public function send($filename, $force_download = TRUE) {
     // Dompdf doesn't have a return value for send so just check the error
     // global it provides.
     if ($errors = $this->getError()) {
-      throw new PrintEngineException(sprintf('Failed to generate PDF: %s', $errors));
+      watchdog_exception('entity_print', new PrintEngineException(sprintf('Failed to generate PDF: %s', $errors)));
     }
 
     // The Dompdf library internally adds the .pdf extension so we remove it
new file mode 100644
