diff --git a/src/Form/AddForm.php b/src/Form/AddForm.php
index 34db4b7..1750dc6 100644
--- a/src/Form/AddForm.php
+++ b/src/Form/AddForm.php
@@ -11,7 +11,7 @@ use Drupal\Core\Form\FormStateInterface;
 use Drupal\favorites\FavoriteStorage;
 use Drupal\Core\Url;
 use Drupal\Component\Utility\UrlHelper;
-use Drupal\Core\StringTranslation\StringTranslationTrait;
+
 
 /**
  * Class AddForm.
@@ -20,8 +20,6 @@ use Drupal\Core\StringTranslation\StringTranslationTrait;
  */
 class AddForm extends FormBase {
 
-  use StringTranslationTrait;
-
   protected $account;
 
   public function __construct(){
