diff --git a/modules/contrib/group/src/Plugin/GroupContentEnabler/GroupMembership.php b/modules/contrib/group/src/Plugin/GroupContentEnabler/GroupMembership.php
index 4e6b4f8..68338f1 100644
--- a/modules/contrib/group/src/Plugin/GroupContentEnabler/GroupMembership.php
+++ b/modules/contrib/group/src/Plugin/GroupContentEnabler/GroupMembership.php
@@ -23,7 +23,7 @@
  *   description = @Translation("Adds users to groups as members."),
  *   entity_type_id = "user",
  *   pretty_path_key = "member",
- *   reference_label = @Translation("Username"),
+ *   reference_label = @Translation("Name"),
  *   reference_description = @Translation("The name of the user you want to make a member"),
  *   enforced = TRUE
  * )
