diff --git a/modules/contrib/group/src/Entity/Group.php b/modules/contrib/group/src/Entity/Group.php
index 8801427..72ea796 100644
--- a/modules/contrib/group/src/Entity/Group.php
+++ b/modules/contrib/group/src/Entity/Group.php
@@ -187,9 +187,12 @@ public function addMember(AccountInterface $account, $values = []) {
    * {@inheritdoc}
    */
   public function getMemberRequestStatus(AccountInterface $account) {
-    $group_content = $this->membershipLoader()->load($this, $account);
-    $data = $group_content->getGroupContent();
-    return $data->values['request_status']['x-default'];
+    $filters = ['entity_id' => $account->id()];
+    $group_content = $this->groupContentStorage()->loadByGroup($this, 'group_membership', $filters);
+    if($group_content){
+      return (int)array_values($group_content)[0]->get('request_status')->value;
+    }
+    return false;
   }
 
   /**
diff --git a/modules/contrib/group/src/Form/GroupRequestMembershipForm.php b/modules/contrib/group/src/Form/GroupRequestMembershipForm.php
index b5fc4bc..180dc23 100644
--- a/modules/contrib/group/src/Form/GroupRequestMembershipForm.php
+++ b/modules/contrib/group/src/Form/GroupRequestMembershipForm.php
@@ -10,17 +10,25 @@
 use Drupal\group\Entity\Form\GroupContentForm;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
+use Drupal\Core\Entity\ContentEntityForm;
 
 /**
  * Provides a Request form for joining a group.
  */
-class GroupRequestMembershipForm extends GroupContentForm {
+class GroupRequestMembershipForm extends ContentEntityForm {
 
   /**
    * {@inheritdoc}
    */
   public function form(array $form, FormStateInterface $form_state) {
     $form = parent::form($form, $form_state);
+    $config = $this->getEntity()->getContentPlugin()->getConfiguration();
+    if (!empty($config['data']['info_text']['value'])) {
+      $form['info_text'] = [
+        '#markup' => $config['data']['info_text']['value'],
+        '#weight' => -99,
+      ];
+    }
     $form['entity_id']['#access'] = FALSE;
     $form['group_roles']['#access'] = FALSE;
     $form['message'] = [
@@ -30,6 +38,14 @@ public function form(array $form, FormStateInterface $form_state) {
     ];
     return $form;
   }
+  
+  function validateForm(array &$form, FormStateInterface $form_state){
+    $group = \Drupal::routeMatch()->getParameter('group');
+    if($group->getMemberRequestStatus(\Drupal::currentUser()) === 0){
+      $form_state->setErrorByName('message','You already have a pending request.');
+    }
+    return parent::validateForm($form, $form_state);
+  }
 
   /**
    * {@inheritdoc}
