Index: src/Form/GroupRequestMembershipForm.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Form/GroupRequestMembershipForm.php	(revision 3a725eb6629d8920ecd994a9b65eb1e912112db0)
+++ src/Form/GroupRequestMembershipForm.php	(revision )
@@ -36,6 +36,7 @@
       '#type' => 'textarea',
       '#title' => $this->t('Message Body'),
       '#description' => $this->t('We will attache this message in request membership mail.'),
+      '#required'=>1,
     ];
     return $form;
   }
