Index: src/GroupMembershipLoader.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/GroupMembershipLoader.php	(revision 1a526d2c8af902639d8e8080b7ef4eff17a97735)
+++ src/GroupMembershipLoader.php	(revision )
@@ -68,7 +68,7 @@
    * {@inheritdoc}
    */
   public function load(GroupInterface $group, AccountInterface $account) {
-    $filters = ['entity_id' => $account->id()];
+    $filters = ['entity_id' => $account->id(),'request_status'=>1];
     $group_contents = $this->groupContentStorage()->loadByGroup($group, 'group_membership', $filters);
     $group_memberships = $this->wrapGroupContentEntities($group_contents);
     return $group_memberships ? reset($group_memberships) : FALSE;
