diff --git a/modules/contrib/printable/printable.module b/modules/contrib/printable/printable.module
index 356b055..2d909d0 100644
--- a/modules/contrib/printable/printable.module
+++ b/modules/contrib/printable/printable.module
@@ -5,7 +5,7 @@
  * Provides printer friendly content entities.
  */
 
-use Drupal\Component\Utility\String;
+//use Drupal\Component\Utility\String;
 use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\Core\Template\Attribute;
@@ -145,7 +145,7 @@ function printable_entity_view_mode_info_alter(&$view_modes) {
 /**
  * Implements hook_entity_view().
  */
-function printable_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode, $langcode) {
+function printable_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
 
   $config = \Drupal::config('printable.settings');
   \Drupal::service('renderer')->addCacheableDependency($build, $config);
