diff --git a/printable.module b/printable.module
index 356b055..c7a5811 100644
--- a/printable.module
+++ b/printable.module
@@ -145,7 +145,7 @@ function printable_entity_view_mode_info_alter(&$view_modes) {
 /**
  * Implements hook_entity_view().
  */
-function printable_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode, $langcode) {
+function printable_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
 
   $config = \Drupal::config('printable.settings');
   \Drupal::service('renderer')->addCacheableDependency($build, $config);
