diff --git a/themes/bootstrap/src/Plugin/Preprocess/Breadcrumb.php b/themes/bootstrap/src/Plugin/Preprocess/Breadcrumb.php
index ae38599..3edc4eb 100644
--- a/themes/bootstrap/src/Plugin/Preprocess/Breadcrumb.php
+++ b/themes/bootstrap/src/Plugin/Preprocess/Breadcrumb.php
@@ -5,8 +5,6 @@
 use Drupal\bootstrap\Utility\Variables;
 use Drupal\Core\Template\Attribute;
 use Drupal\Core\Url;
-use Drupal\Component\Utility\Html;
-
 /**
  * Pre-processes variables for the "breadcrumb" theme hook.
  *
@@ -46,7 +44,7 @@ public function preprocessVariables(Variables $variables) {
       $page_title = \Drupal::service('title_resolver')->getTitle($request, $route_match->getRouteObject());
       if (!empty($page_title)) {
         $breadcrumb[] = [
-          'text' => \Drupal\Core\Render\Markup::create(Html::decodeEntities($page_title)),
+          'text' => $page_title,
           'attributes' => new Attribute(['class' => ['active']]),
         ];
       }
