diff --git a/modules/contrib/group/src/Plugin/views/argument/GroupId.php b/modules/contrib/group/src/Plugin/views/argument/GroupId.php
index b79895f55..8c2960f61 100644
--- a/modules/contrib/group/src/Plugin/views/argument/GroupId.php
+++ b/modules/contrib/group/src/Plugin/views/argument/GroupId.php
@@ -56,6 +56,8 @@ public function titleQuery() {
     $titles = [];
 
     $groups = $this->groupStorage->loadMultiple($this->value);
+    $values = is_array($this->value[0])?$this->value[0]:$this->value;
+    $groups = $this->groupStorage->loadMultiple($values);
     foreach ($groups as $group) {
       $titles[] = $group->label();
     }
