diff --git a/themes/bootstrap/js/misc/tabledrag.js b/themes/bootstrap/js/misc/tabledrag.js
index ac41c8f6b..99b67169b 100644
--- a/themes/bootstrap/js/misc/tabledrag.js
+++ b/themes/bootstrap/js/misc/tabledrag.js
@@ -121,6 +121,12 @@
      */
     this.windowHeight = 0;
 
+    /**
+     *
+     * @type {?HTMLElement}
+     */
+     this.$toggleWeightButton = null;
+
     /**
      * Check this table's settings to see if there are parent relationships in
      * this table. For efficiency, large sections of code can be skipped if we
@@ -175,9 +181,11 @@
     // Add a link before the table for users to show or hide weight columns.
     var $button = $(Drupal.theme('btn-sm', {
       'class': ['tabledrag-toggle-weight'],
+      'data-drupal-selector': ['tabledrag-toggle-weight'],
       title: Drupal.t('Re-order rows by numerical weight instead of dragging.'),
       'data-toggle': 'tooltip'
     }));
+    this.$toggleWeightButton = $('[data-drupal-selector="tabledrag-toggle-weight"]');
 
     $button
       .on('click', $.proxy(function (e) {
diff --git a/themes/contrib/bootstrap/js/misc/tabledrag.js b/themes/contrib/bootstrap/js/misc/tabledrag.js
index ac41c8f6b..75326e310 100644
--- a/themes/contrib/bootstrap/js/misc/tabledrag.js
+++ b/themes/contrib/bootstrap/js/misc/tabledrag.js
@@ -2,7 +2,7 @@
  * @file
  * Extends methods from core/misc/tabledrag.js.
  */
-(function ($) {
+ (function ($) {
 
   // Save the original prototype.
   var prototype = Drupal.tableDrag.prototype;
@@ -121,6 +121,11 @@
      */
     this.windowHeight = 0;
 
+    /**
+     * @type {?HTMLElement}
+     */
+    this.$toggleWeightButton = null;
+
     /**
      * Check this table's settings to see if there are parent relationships in
      * this table. For efficiency, large sections of code can be skipped if we
@@ -173,21 +178,24 @@
     $table.find('> tr.draggable, > tbody > tr.draggable').each(function () { self.makeDraggable(this); });
 
     // Add a link before the table for users to show or hide weight columns.
-    var $button = $(Drupal.theme('btn-sm', {
+    self.$toggleWeightButton = $(Drupal.theme('btn-sm', {
       'class': ['tabledrag-toggle-weight'],
+      'data-drupal-selector': ['tabledrag-toggle-weight'],
       title: Drupal.t('Re-order rows by numerical weight instead of dragging.'),
       'data-toggle': 'tooltip'
     }));
 
-    $button
+    self.$toggleWeightButton = $('[data-drupal-selector="tabledrag-toggle-weight"]');
+
+    self.$toggleWeightButton
       .on('click', $.proxy(function (e) {
         e.preventDefault();
         this.toggleColumns();
       }, this))
       .wrap('<div class="tabledrag-toggle-weight-wrapper"></div>')
-      .parent()
-    ;
-    $table.before($button);
+      .parent();
+
+    $table.before(self.$toggleWeightButton);
 
     // Initialize the specified columns (for example, weight or parent columns)
     // to show or hide according to user preference. This aids accessibility
