diff --git a/core/modules/user/src/AccountForm.php b/core/modules/user/src/AccountForm.php
index 4ac72110b..26395c6fe 100644
--- a/core/modules/user/src/AccountForm.php
+++ b/core/modules/user/src/AccountForm.php
@@ -67,7 +67,7 @@ public function form(array $form, FormStateInterface $form_state) {
     $user = $this->currentUser();
     $config = \Drupal::config('user.settings');
     $form['#cache']['tags'] = $config->getCacheTags();
-
+    $admin = $user->hasPermission('administer users');
     $language_interface = \Drupal::languageManager()->getCurrentLanguage();
 
     // Check for new account.
@@ -115,7 +115,7 @@ public function form(array $form, FormStateInterface $form_state) {
         'spellcheck' => 'false',
       ],
       '#default_value' => (!$register ? $account->getAccountName() : ''),
-      '#access' => $account->name->access('edit'),
+      '#access' => ($register || ($user->id() == $account->id() && $user->hasPermission('change own username')) || $admin),
     ];
 
     // Display password field only for existing users or when user is allowed to
