diff --git a/core/lib/Drupal/Core/Routing/UrlGenerator.php b/core/lib/Drupal/Core/Routing/UrlGenerator.php
index f56736ab2..018e4cbb4 100644
--- a/core/lib/Drupal/Core/Routing/UrlGenerator.php
+++ b/core/lib/Drupal/Core/Routing/UrlGenerator.php
@@ -198,12 +198,18 @@ protected function doGenerate(array $variables, array $defaults, array $tokens,
       if ('variable' === $token[0]) {
         if (!$optional || !array_key_exists($token[3], $defaults) || (isset($mergedParams[$token[3]]) && (string) $mergedParams[$token[3]] !== (string) $defaults[$token[3]])) {
           // check requirement
-          if (!preg_match('#^' . $token[2] . '$#', $mergedParams[$token[3]])) {
-            $message = sprintf('Parameter "%s" for route "%s" must match "%s" ("%s" given) to generate a corresponding URL.', $token[3], $name, $token[2], $mergedParams[$token[3]]);
+	  if (is_object($mergedParams[$token[3]])) {
+             $nid = $mergedParams[$token[3]]->id();
+           }
+	   else {
+             $nid = $mergedParams[$token[3]];
+           }
+  	   if (!preg_match('#^' . $token[2] . '$#', $nid)) {
+            $message = sprintf('Parameter "%s" for route "%s" must match "%s" ("%s" given) to generate a corresponding URL.', $token[3], $name, $token[2], $nid);
             throw new InvalidParameterException($message);
           }
 
-          $url = $token[1] . $mergedParams[$token[3]] . $url;
+          $url = $token[1] . $nid . $url;
           $optional = FALSE;
         }
       }
diff --git a/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php b/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php
index dab14ef01..1d93381a1 100644
--- a/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php
+++ b/core/modules/editor/src/Plugin/InPlaceEditor/Editor.php
@@ -30,7 +30,7 @@ public function isCompatible(FieldItemListInterface $items) {
     // This editor is compatible with formatted ("rich") text fields; but only
     // if there is a currently active text format, that text format has an
     // associated editor and that editor supports inline editing.
-    elseif ($editor = editor_load($items[0]->format)) {
+    elseif ($editor = !empty($items[0]->format)?editor_load($items[0]->format):'') {
       $definition = \Drupal::service('plugin.manager.editor')->getDefinition($editor->getEditor());
       if ($definition['supports_inline_editing'] === TRUE) {
         return TRUE;
