diff --git a/core/lib/Drupal/Core/Form/FormValidator.php b/core/lib/Drupal/Core/Form/FormValidator.php
index 57d24cc21..3e6c8ab1f 100644
--- a/core/lib/Drupal/Core/Form/FormValidator.php
+++ b/core/lib/Drupal/Core/Form/FormValidator.php
@@ -330,7 +330,12 @@ protected function doValidateForm(&$elements, FormStateInterface &$form_state, $
    */
   protected function performRequiredValidation(&$elements, FormStateInterface &$form_state) {
     // Verify that the value is not longer than #maxlength.
-    if (isset($elements['#maxlength']) && mb_strlen($elements['#value']) > $elements['#maxlength']) {
+    $length = 0;
+    if (!isset($elements['#value'])) {
+      $length = count($elements['#value']) > 0 ? mb_strlen(str_replace(array("\n", "\r\n", "\r"), '', $elements['#value'])): 0;
+    }
+
+    if (isset($elements['#maxlength']) && $length > $elements['#maxlength']) {
       $form_state->setError($elements, $this->t('@name cannot be longer than %max characters but is currently %length characters long.', ['@name' => empty($elements['#title']) ? $elements['#parents'][0] : $elements['#title'], '%max' => $elements['#maxlength'], '%length' => mb_strlen($elements['#value'])]));
     }
 
