diff --git a/modules/contrib/subpathauto/src/PathProcessor.php b/modules/contrib/subpathauto/src/PathProcessor.php
index 9b2eb5245..9af79620c 100644
--- a/modules/contrib/subpathauto/src/PathProcessor.php
+++ b/modules/contrib/subpathauto/src/PathProcessor.php
@@ -157,7 +157,7 @@ protected function getPath($path_info) {
       $path_info = '/' . substr($path_info, strlen($language_prefix));
     }
 
-    return $path_info;
+    return rtrim(urldecode($path_info),'/');
   }
 
   /**
